flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/FileSourceFunction.java

```diff
@@ -123,9 +127,11 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 			OUT nextElement = serializer.createInstance();
 			nextElement =  format.nextRecord(nextElement);
 			if (nextElement == null && splitIterator.hasNext()) {
+				splitCounter.inc();
```

Comment:
In `DataSourceTask` the counter is incremented when a split is closed. We should sync both classes.